### PR TITLE
OTEL_EXPORTER_OTLP_PROTOCOL autoconfiguration

### PR DESCRIPTION
**Pull request by chameleon82**
This PR to support next env auto-configurations:

https://opentelemetry.io/docs/concepts/sdk-configuration/otlp-exporter-configuration/#otel_exporter_otlp_protocol

with `http/protobuf` as default
```
OTEL_EXPORTER_OTLP_PROTOCOL=grpc
OTEL_EXPORTER_OTLP_PROTOCOL=http/protobuf
OTEL_EXPORTER_OTLP_PROTOCOL=http/json
```
with overrides:
```
OTEL_EXPORTER_OTLP_LOGS_PROTOCOL=grpc
OTEL_EXPORTER_OTLP_LOGS_PROTOCOL=http/protobuf
OTEL_EXPORTER_OTLP_LOGS_PROTOCOL=http/json
```

### CODE REVIEW

File: exporters/otlp/otlplogs/otlplogshttpjson/client.go [hunk 1]
```diff
@@ -341,3 +341,7 @@ func (d *client) MarshalLog() interface{} {
 		Insecure: d.cfg.Insecure,
 	}
 }
+
+func init() {
```
Comment (chameleon82) [inline exporters/otlp/otlplogs/otlplogshttpjson/client.go:345]: Init function in all-in-one module will be executed once, only one of the modules will be registered this way.
this solution will not work

File: exporters/otlp/otlplogs/otlplogshttp/client.go [hunk 2]
```diff
@@ -339,3 +339,7 @@ func (d *client) MarshalLog() interface{} {
 		Insecure: d.cfg.Insecure,
 	}
 }
+
+func init() {
+	otlplogs.Clients[otlpconfig.ExporterProtocolHttpProtobuf] = NewClient()
+}
```
Comment (natebwangsut) [inline exporters/otlp/otlplogs/otlplogshttp/client.go:345]: Same goes to here as well, the registration will be executed only once on the startup

File: exporters/otlp/otlplogs/otlplogsgrpc/client.go [hunk 3]
```diff
@@ -299,3 +299,7 @@ func (c *client) MarshalLog() interface{} {
 		Endpoint: c.endpoint,
 	}
 }
+
+func init() {
+	otlplogs.Clients[otlpconfig.ExporterProtocolGrpc] = NewClient()
```
Comment (natebwangsut) [inline exporters/otlp/otlplogs/otlplogsgrpc/client.go:304]: and here

**Comment by natebwangsut** [comment]
other than that, this is looking good, the New() function has been refactored to be more go-like

**Pull request merged**